Tests added

Cammelen opened the pull request.

Commit:
Tests added

nikolay-977 commented on pom.xml:
```diff
@@ -0,0 +1,110 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Можно улучшить: хорошей практикой является добавление README.md с описанием проекта. Как заполнять README.md можно ознакомиться в статье https://codengineering.net/post/26943

nikolay-977 commented on src/test/java/CreateCourierTests.java:
```diff
@@ -0,0 +1,99 @@
+
+    @Test
+    @DisplayName("Успешное создание нового курьера")
+    public void createCourier() {
```
Необходимо исправить: Используй инкапсуляцию. В src.main.java.api.client для каждой ручки создай классы CourierClient(для /api/v1/courier) и OrdersClient(для /api/v1/orders). Вынеси реализацию шагов в эти классы, а именно - в классах CourierClient и OrdersClient реализуй методы по примеру:

```
@Step("Get response for incorrect password") 
public Response getIncorrectPasswordRsponse(Object body) {
    return given()
    .header("Content-type", "application/json")
    .and()
    .body(body)
    .when()
    .post(loginCourier);
    .then();
}
```

И приведи кейсы к виду:

```
@Test 
@DisplayName("Check error message for incorrect password") 
public void testErrorMessageForIncorrectPassword(){ 
    CourierClient courierClient = new CourierClient();
    Response incorrectPasswordRsponse = courierClient.getIncorrectPasswordRsponse(new 
    AuthCourier(currentCourier.login,"*** WrongPassword ***"));
    incorrectPasswordRsponse.statusCode(404).and.assertThat().body("message", is("Учетная запись не найдена"));
}
```

Это позволит избежать повторяющегося кода, а также переиспользовать шаги в разных тестах, что сократит количество кода автотестов, улучшит читаемость и упростит поддержку.

nikolay-977 commented on src/test/java/CreateCourierTests.java:
```diff
@@ -0,0 +1,99 @@
+
+    private final Steps steps = new Steps();
+
+    Courier courier = new Courier("Jako" + new Random().nextInt(1000), "1111", "Jan");
```
Необходимо исправить: Для инстанс-переменных следует указывать модификатор доступа private.

nikolay-977 commented on src/test/java/CreateCourierTests.java:
```diff
@@ -0,0 +1,99 @@
+                .post("/api/v1/courier");
+        steps.checkStatusCode201(response);
+        response.then().assertThat().body("ok", equalTo(true));
+        System.out.println(courier.getLogin());
```
Необходимо исправить: Проверки должны быть в конце тестовых методов. Между проверками и после проверок не должно быть дополнительных действий. Иначе в случае фейла первой проверки, последующие действия(шаги и методы) не будут выполнены, это может аффектить другие тесты, что увеличит время локализации дефекта.

nikolay-977 commented on src/test/java/CreateCourierTests.java:
```diff
@@ -0,0 +1,99 @@
+                .header("Content-type", "application/json")
+                .auth().oauth2(bearerToken)
+                .body(courier)
+                .post("/api/v1/courier");
```
Необходимо исправить: следует избегать повторяющегося кода, uri лучше вынести в константы

nikolay-977 commented on src/test/java/Steps.java:
```diff
@@ -0,0 +1,42 @@
+import org.junit.After;
+import org.junit.Before;
+
+public class Steps {
```
Класс лучше переименовать в BaseTest и унаследовать от него тестовые классы

nikolay-977 commented on src/test/java/Steps.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    @After
+    public void cleanUp() {
```
Необходим исправить: данный метод имеет пустое тело. Его следует использовать для удаления тестовых данных.

nikolay-977 commented on src/test/java/Steps.java:
```diff
@@ -0,0 +1,42 @@
+
+public class Steps {
+
+    String bearerToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI2M2EwNTlmM2QzYjg2YTAwM2Q2ODJmNzYiLCJpYXQiOjE2NzY0NzY5OTgsImV4cCI6MTY3NzA4MTc5OH0.eIBvdlijIzWe0giUUWy2hy2C6CHgjb4MGEmnLYxep5E";
```
Необходимо исправить: Для инстанс-переменных следует указывать модификатор доступа.

Commit:
Tests fixed

nikolay-977 commented on src/test/java/CreateCourierTests.java:
```diff
@@ -0,0 +1,94 @@
+    public void createExistCourier() {
+        courierClient.checkCreateCourierApi(courier);
+        Response response = courierClient.checkCreateCourierApi(courier);
+        System.out.println(courier.getLogin());
```
Можно улучшить: если System.out.println() используется для отладки, лучше использовать логирование. https://www.baeldung.com/java-logging-intro

nikolay-977 commented on src/test/java/ExistingCourierTests.java:
```diff
@@ -0,0 +1,73 @@
+        System.out.println(existingCourierOnceLogin.getPassword());
+        steps.checkStatusCode400(response);
+        steps.printResponseBody(response);
+        response.then().assertThat().body("message", equalTo("Недостаточно данных для входа"));
```
Необходимо исправить: проверки в тестовых методах должны быть в одном стиле. Когда код написан в одном стиле, это улучшает читаемость кода, сокращает время локализации дефекта и упрощает его поддержку. Не следует в тестовых методах одновременно использовать инкапсулированные проверки (вынесенные в BaseTest) и проверки реализованные внутри тестовых метода. Также не очень хорошая практика -  выносить проверки в один класс(BaseTest), т.к в случае увеличения количества тестов, класс будет очень большим. Лучше выбрать единный подход - либо все проверки вынести в слой шагов, либо реализовать все проверки внутри тестовых методов, что сократит количество строчек кода.

Commit:
Исправлено:

1. Во всех тестах убран System.out.println()
2. Проверки из BaseTest перенесены в сами тесты
